app/routes/claim/view-claim.js

```diff
@@ -39,25 +39,9 @@ module.exports = function (router) {
       if (error instanceof ValidationError) {
         getIndividualClaimDetails(req.params.claimId)
           .then(function (data) {
-            // TODO move to route helper and test
-
             if (data.claim && data.claimExpenses) {
               data.claim.NomisCheck = req.body.nomisCheck
-
-              var claimExpensesById = {}
-              claimExpenses.forEach(function (claimExpense) {
-                claimExpensesById[claimExpense.claimExpenseId] = claimExpense
-              })
-              data.claimExpenses.forEach(function (expense) {
-                var postedClaimExpenseResponse = claimExpensesById[expense.ClaimExpenseId.toString()]
-                expense.Status = postedClaimExpenseResponse.status
-                if (expense.Status === 'APPROVED-DIFF-AMOUNT') {
-                  expense.ApprovedCost = postedClaimExpenseResponse.approvedCost
-                  if (!Validator.isCurrency(postedClaimExpenseResponse.approvedCost)) {
-                    expense.Error = true
-                  }
-                }
-              })
+              data.claimExpenses = getNonPersistedExpenseData(data.claimExpenses, claimExpenses)
```

Comment:
Don't like the name, doesn't describe what it does.
mergeClaimExpensesWithSubmittedResponses